### PR TITLE
feat: add support for `Bytes` and `RawSlice` as inputs and outputs for Contracts

### LINKED ISSUE #1188: Add support for the new stdlib type `Bytes` and `RawSlice` for inputs and outputs for Contracts
### Motivation

https://github.com/FuelLabs/fuels-rs/issues/723

### Usage example

See #933

### Possible implementations

https://github.com/FuelLabs/fuels-rs/pull/868

### PR DESCRIPTION
**Pull request by camsjams**
closes #1188 

### Bytes
vector of u8 data

### Raw Slice
vector of u64 data

### CODE REVIEW

**Review by nedsalk — DISMISSED** [review]
LGTM! Although you've got a failing test, so another review round will be needed. But I don't see any issue with the code as it is.

File: packages/abi-coder/src/coders/byte.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { arrayify, concat, type BytesLike } from '@ethersproject/bytes';
```
Comment (nedsalk) [inline packages/abi-coder/src/coders/byte.ts:1]:
```suggestion
import { arrayify, concat } from '@ethersproject/bytes';
```

The linting is saying `BytesLike` is unused.

Comment (nedsalk) [reply packages/abi-coder/src/coders/byte.ts]: I committed it to shorten the review cycle.

**Review by nedsalk — DISMISSED** [review]

**Review by nedsalk — DISMISSED** [review]

**Review by nedsalk — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]
🪄

**Pull request merged**